### PR TITLE
BGDIINF_SB-1746: Removed the Item renaming functionality

### PR DESCRIPTION
**Pull request by ltshb**
Also improved the Asset admin page. In this one the read only item field
still had the help text for the field search functionality which don't
make sense for read only field. To avoid this simply changed the logic,
adding extra read only fields that are only displayed in the change view
and the write item field is only display in the add view.

Also improved the logging in case off renaming.

### CODE REVIEW

File: app/stac_api/admin.py [hunk 1]
```diff
@@ -208,11 +233,11 @@ def get_search_results(self, request, queryset, search_term):
                 queryset &= self.model.objects.filter(item__collection__name__exact=collection_name)
         return queryset, use_distinct
 
-    def collection(self, instance):
+    def collection_name(self, instance):
         return instance.item.collection
 
-    collection.admin_order_field = 'item__collection'
-    collection.short_description = 'Collection Id'
+    collection_name.admin_order_field = 'item__collection'
```
Comment (hansmannj) [inline app/stac_api/admin.py:239]: shouldn't that be `item__collection__name` here?

File: app/stac_api/admin.py [hunk 2]
```diff
@@ -208,11 +233,11 @@ def get_search_results(self, request, queryset, search_term):
                 queryset &= self.model.objects.filter(item__collection__name__exact=collection_name)
         return queryset, use_distinct
 
-    def collection(self, instance):
+    def collection_name(self, instance):
         return instance.item.collection
```
Comment (hansmannj) [inline app/stac_api/admin.py:237]: and probably `instance.item.collection.name` here?

File: app/tests/test_items_endpoint.py [hunk 3]
```diff
@@ -661,19 +665,23 @@ def test_item_endpoint_patch_rename_item(self):
...
-        path = f'/{STAC_BASE_V}/collections/{self.collection["name"]}/items/{data["id"]}'
-        response = self.client.get(path)
-        json_data = response.json()
-        self.assertStatusCode(200, response)
-        self.assertEqual(data['id'], json_data['id'])
+        # Make sure the rename item was done
+        self.assertFalse(
+            Item.objects.all().filter(name=data["id"],
+                                      collection__name=self.collection['name']).exists(),
+            msg="Original item doesn't exists anymore after trying to rename it"
```
Comment (hansmannj) [inline app/tests/test_items_endpoint.py:683]: msg should be something like "Renamed item shouldn't exist" or the like

File: app/tests/test_items_endpoint.py [hunk 4]
```diff
@@ -661,19 +665,23 @@ def test_item_endpoint_patch_rename_item(self):
...
+        )
 
-        # Check the data by reading it back
-        path = f'/{STAC_BASE_V}/collections/{self.collection["name"]}/items/{data["id"]}'
-        response = self.client.get(path)
-        json_data = response.json()
-        self.assertStatusCode(200, response)
-        self.assertEqual(data['id'], json_data['id'])
+        # Make sure the rename item was done
```
Comment (hansmannj) [inline app/tests/test_items_endpoint.py:679]: "Make sure the rename item was NOT done", I guess

File: app/tests/test_items_endpoint.py [hunk 5]
```diff
@@ -601,20 +601,24 @@ def test_item_endpoint_put_rename_item(self):
...
+        )
 
-        # Check the data by reading it back
-        path = f'/{STAC_BASE_V}/collections/{self.collection["name"]}/items/{sample.json["id"]}'
-        response = self.client.get(path)
-        json_data = response.json()
-        self.assertStatusCode(200, response)
-        self.assertEqual(sample.json['id'], json_data['id'])
-        self.check_stac_item(sample.json, json_data, self.collection["name"])
+        # Make sure the rename item was done
```
Comment (hansmannj) [inline app/tests/test_items_endpoint.py:615]: Make sure the rename item was NOT done

File: app/tests/test_items_endpoint.py [hunk 6]
```diff
@@ -601,20 +601,24 @@ def test_item_endpoint_put_rename_item(self):
...
+        self.assertFalse(
+            Item.objects.all().filter(
+                name=sample["name"], collection__name=self.collection['name']
+            ).exists(),
+            msg="Original item doesn't exists anymore after trying to rename it"
```
Comment (hansmannj) [inline app/tests/test_items_endpoint.py:620]: msg should be something like "Renamed item shouldn't exist" or the like

**Review by hansmannj — APPROVED** [review]
:+1:
Only a few minor things

**Review by IsabelleBzr — APPROVED** [review]
:+1:

**Pull request merged**